### PR TITLE
taup: fix colors in ray path plots

### LINKED ISSUE #2470: obspy.taup.tau.Arrivals.plot_rays: color of ray paths
ObsPy version: 1.1.0
Python version: 3.6
Platform: OsX and Anaconda

Hi, 

I'm plotting 3 phases ["PcPPcPPKP", "PcPPKPPcP", "PKPPcPPcP"], which have similar ray path, using obspy.taup.tau.Arrivals.plot_rays. Since they are all plotted in the same color, it's difficult to distinguish them. However, there is no option in that function to change the color of the ray paths. Is there a way to do that?

Raypath_cKK

Thank you in advance.

### PR DESCRIPTION
**Pull request by megies**
### What does this PR do?

Fix colors in ray path plots and properly cycle through our fixed colors. Plots are gonna change colors, so might need updating image test, I guess..

### Why was it initiated?  Any relevant Issues?

Fixes #2470 

### PR Checklist
- [x] Correct base branch selected? `master` for new features, `maintenance_...` for bug fixes
- [x] This PR is not directly related to an existing issue (which has no PR yet).
- [x] If the PR is making changes to documentation, docs pages can be built automatically.
      Just remove the space in the following string after the + sign: "+ DOCS"
- [x] If any network modules should be tested for the PR, add them as a comma separated list
      (e.g. `clients.fdsn,clients.arclink`) after the colon in the following magic string: "+TESTS:"
      (you can also add "ALL" to just simply run all tests across all modules)
- [ ] All tests still pass.
- [x] Any new features or fixed regressions are be covered via new tests.
- [x] Any new or changed features have are fully documented.
- [x] Significant changes have been added to `CHANGELOG.txt` .
- [x] First time contributors have added your name to `CONTRIBUTORS.txt` .

+DOCS

### CODE REVIEW

**Comment by megies** [comment]
@okling would be nice if you could review this change

File: obspy/taup/tau.py [hunk 1]
```diff
@@ -346,17 +346,15 @@ def plot_rays(self, phase_list=None, plot_type="spherical",
 
             intp = matplotlib.cbook.simple_linear_interpolation
             radius = self.model.radius_of_planet
+            phase_names_encountered = set([ray.name for ray in arrivals])
```
Comment (QuLogic) [inline obspy/taup/tau.py]: Can use a set literal.
```suggestion
            phase_names_encountered = {ray.name for ray in arrivals}
```

File: obspy/taup/tau.py [hunk 2]
```diff
@@ -346,17 +346,15 @@ def plot_rays(self, phase_list=None, plot_type="spherical",
...
+            colors = {name: COLORS[i % len(COLORS)]
+                      for i, name in enumerate(phase_names_encountered)}
```
Comment (QuLogic) [inline obspy/taup/tau.py]: This probably won't be sorted, so it might not be consistent.

File: obspy/taup/tau.py (hunk 1 above)
Comment (megies) [reply obspy/taup/tau.py]: Oh OK, didn't realize that was in 2.7

File: obspy/taup/tau.py (hunk 2 above)
Comment (megies) [reply obspy/taup/tau.py]: Good catch

File: obspy/taup/tau.py [hunk 3]
```diff
@@ -346,17 +346,16 @@ def plot_rays(self, phase_list=None, plot_type="spherical",
 
             intp = matplotlib.cbook.simple_linear_interpolation
             radius = self.model.radius_of_planet
+            phase_names_encountered = {ray.name for ray in arrivals}
```
Comment (QuLogic) [inline obspy/taup/tau.py]: I wonder if we should set `colors` from `phase_names` first, then add remaining encountered ones. This way you can call `plot_rays` multiple times on the same `Axes` and get consistent colours for stuff you explicitly asked for.

Comment (megies) [reply obspy/taup/tau.py]: Done

**Comment by megies** [comment]
CI passes, gonna do a final rebase and merge

**Pull request merged**